helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterModelProvider.java

```diff
@@ -53,7 +54,31 @@ private enum RebalanceScopeType {
     // changes.
     GLOBAL_BASELINE,
     // Set the rebalance scope to cover only replicas that are assigned to downed instances.
-    EMERGENCY
+    EMERGENCY,
+    // A temporary overwrites for partition replicas on downed instance but still within the delayed window but missing
+    // minActiveReplicas
+    DELAYED_REBALANCE_OVERWRITES
+  }
+
+  /**
+   * TODO: On integration with WAGED, have to integrate with counter and latency metrics -- qqu
+   * Compute a new Cluster Model with scope limited to partitions with best possible assignment missing minActiveReplicas
+   * because of delayed rebalance setting.
+   * @param dataProvider The controller's data cache
+   * @param resourceMap The full map of the resource by name
+   * @param activeInstances The active instances that will be used in the calculation.
+   * @param resourceAssignment The resource assignment state to compute on. This should be the current state assignment;
+   *                           if it's run right after another rebalance calculation, the best possible assignment from
+   *                           previous result can be used.
+   * @return the ClusterModel
+   */
+  public static ClusterModel generateClusterModelForDelayedRebalanceOverwrites(
```

Comment:
I guess the hooking of the call will be part 2, right?

But logically, I am not sure whether this is the right place, the hierarchy is:

rebalance -> Baseline
                  -> Partial
                  -> Emergency
                  
                  
But delay caused overrides sounds more part of partial rebalance. If we have a separate entry here could be wired.